pkg/tfbridge/info.go

```diff
@@ -492,6 +492,12 @@ type ComputeDefaultOptions struct {
 	// example, that random values generated across "pulumi preview" and "pulumi up" in the same deployment are
 	// consistent. This currently is only available for resource changes.
 	Seed []byte
+
+	// Path to the sub-property where the default application is happening. For example,
+	// resource.PropertyPath{"propName"} indicates applying defaults to a top-level property, and
+	// resource.PropertyPath{"propName", 1, "subProp"} indicates applying defaults to the "subProp" property of the
+	// second element of the array found under "propName".
+	PropertyPath resource.PropertyPath
```

Comment:
>  Users (including SetAutoNaming) already know which field is being targeted

Have you seen this block?

```
	// Reuse the value from prior state if available. Note that this code currently only runs for Plugin Framework
	// resources, as SDKv2 based resources avoid calling ComputedDefaults in the first place in update situations.
	// To do that SDKv2 based resources track __defaults meta-key to distinguish between values originating from
	// defaulting machinery from values originating from user code. Unfortunately Plugin Framework cannot reliably
	// disinguish default values, therefore it always calls ComputedDefaults. To compensate, this code block avoids
	// re-generating the auto-name if it is located in PriorState and reuses the old one; this avoids generating a
	// fresh random value and causing a replace plan.
	if defaultOptions.PriorState != nil && defaultOptions.PropertyPath != nil {
		prior := resource.NewObjectProperty(defaultOptions.PriorState)
		oldV, gotOldV := defaultOptions.PropertyPath.Get(prior)
		if oldV.IsString() && gotOldV {
			return oldV.StringValue(), nil
		}
	}
```

As written, when using `AutoName(name string, maxlength int, separator string) *SchemaInfo` this context is lost. In the case of SetAutoNaming one can assume `PropertyPath{"name"}` but currently AutoName is theoretically usable at any nested position and can be used on properties other than "name". It needs to be able to find the prior state value.

Reply:
I'm noticing MakeTerraformInput has  `old` resource.PropertyValue tracked. In that case we could pass in the old value instead of the path. This should work just as well for this purpose.